server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -211,8 +211,9 @@ public DateTime(StreamInput in) throws IOException {
                 this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
             }
             final boolean isJoda;
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                //if stream is from 7.7 Node it will have a flag indicating if format is joda
+            if (in.getVersion().onOrAfter(Version.V_7_7_0) && in.getVersion().major == 7) {
```

Comment:
What is this condition trying to accomplish? The major version of 7.7.0 is already 7, so it seems redundant

Reply:
I see the intent now. I agree with @benwtrent's assessment; backcompat should be handled from the future version, not here.